src/models/krate.rs

```diff
@@ -354,74 +355,51 @@ impl Crate {
         Ok(users.chain(teams).collect())
     }
 
-    /// Invite `login` as an owner of this crate, returning a status message and
-    /// optionally an invite email to be sent by the caller.
+    /// Invite `login` as an owner of this crate, returning the created
+    /// [`NewOwnerInvite`].
     pub fn owner_add(
```

Comment:
it looks like the controller is the only user of this fn. I'm wondering if it's even worth keeping it on the model at all... if we moved/inlined it completely we might not even need the new enums.

Reply:
I started to pull this out, but there's a decent chunk of database-specific / diesel query building bits in the "team" path:

https://github.com/domodwyer/crates.io/blob/0b0db20b529cbecdf8ff31d022ead3da2c9ecf50/src/models/krate.rs#L388-L400

I'd be inclined to keep the model fn because of it? The fn does abstract the caller from "what kind of thing are you adding" which is worthwhile too.